guide/jupyter_guide.adoc

```diff
@@ -0,0 +1,225 @@
+
+====== Sample Script
+The script below will be what will be used with the sample Jupyter Notebook already provided. This script should work for most standard python3 notebooks provided that the *parameters* dictionary is given the correct pairs.
+
+```python
+include::example/jupyter/run_notebook.py[]
+```
+
+If it is necessary to create a custom script for your Jupyter Notebook, the following will go more in depth on the general process of how to make a script.
+
+====== Reading in the Notebook
+Reading in a notebook requires first opening up a notebook and then reading the json data that makes up the notebook to a variable, which in this case will be named *nb*. Opening the notebook is standard, reading it uses the *nbformat* tool. This will also be used later to write to an output notebook. Code to open a notebook is shown below.
+
+```python
+f = open(nbFileName, "r")
+nb = nbformat.read(f, as_version=4)
+f.close()
+```
+
+====== Papermill
+As previously discussed, to add parameters to a Jupyter Notebook a "parameter" tag needs to be attached to a cell so that Papermill recognizes it. An example of how Papermill can be used is below.
+
+```python
+nb = mill.execute_notebook(
+      nb_input_file_path,
+      nb_output_file_path,
+      parameters=dict(<parameter>=<value>, ...),
+      prepare_only=True,
+      log_output=True
+    )
+```
+
+*nb_input_file_path* and *nb_output_file_path* are the direct paths to the intended location and names of the input and output Jupyter Notebooks. The `parameters` parameter must be a dict containing pairs of parameters set in the Jupyter Notebook and their intended value. *prepare_only=true* only exports the parameter values to the Jupyter Notebook and does not execute the notebook. Execution of the notebook is done later in the script using Jupyter's *nbconvert* tool.
+
+====== Preprocessor
+While its not entirely necessary to manually create a preprocessor to run the notebook, but doing so gives you a lot more customization including:
+
+* Cell-by-cell execution
+* Specifying the Kernel used 
+* Custom Preprocessor
+
+Creating a custom preprocessor will require development, but will give the most control over execution. Cell-by-cell execution and specifying the kernel are easy to implement by comparison and are used in the sample script. To create a preprocessor to execute the notebook, look at the code displayed below.
+
+```python
+ex = ExecutePreprocessor(kernel_name=<kernel>)
+ex.km, ex.kc = ex.start_new_kernel()
+```
+
+For the value of *kernel_name* set it to python2 or python3, depending on what version used to make the Jupyter Notebook. The python kernel is already required for any notebook, regardless of kernel, so nothing else extra is required.
```

Comment:
We are using `python3` everywhere else, should we remove the reference to `python2`?